DataFormats/Detectors/MUON/MID/src/Track.cxx

```diff
@@ -82,14 +78,14 @@ bool Track::propagateToZ(float zPosition)
   /// A linear extraplation is performed
 
   // Nothing to be done if we're already at zPosition
-  if (mPosition.z() == zPosition) {
+  if (mPosition[2] == zPosition) {
```

Comment:
strict equality for floats ? Is this really what you want/need ?

Reply:
When I extrapolate I do it at the z of the cluster which is at a well defined z position (expressed in floating precision).
So, precise equality is actually what I want.